include/deal.II/base/config.h.in

```diff
@@ -390,41 +390,27 @@
 /*
  * PETSc:
  *
- * Note: The following definitions will be set in petscconf.h and
- *       petscversion.h, so we don't repeat them here.
- *
- *  PETSC_VERSION_MAJOR
- *  PETSC_VERSION_MINOR
- *  PETSC_VERSION_SUBMINOR
- *  PETSC_VERSION_PATCH
- *  PETSC_VERSION_RELEASE
- *  PETSC_USE_COMPLEX
- */
-
-#define DEAL_II_PETSC_VERSION_LT(major,minor,subminor) \
-  ((PETSC_VERSION_MAJOR * 10000 + \
-    PETSC_VERSION_MINOR * 100 + \
-    PETSC_VERSION_SUBMINOR) \
-    <  \
-    (major)*10000 + (minor)*100 + (subminor))
+ * Note: The following macros are defined in petscversion.h
+ *       so we don't repeat them here.
+ */
 
-#define DEAL_II_PETSC_VERSION_GTE(major,minor,subminor) \
-  ((PETSC_VERSION_MAJOR * 10000 + \
-    PETSC_VERSION_MINOR * 100 + \
-    PETSC_VERSION_SUBMINOR) \
-    >=  \
-    (major)*10000 + (minor)*100 + (subminor))
+#ifdef DEAL_II_WITH_PETSC
+#  define DEAL_II_PETSC_VERSION_LT(major,minor,subminor) \
+    PETSC_VERSION_LT(major,minor,subminor)
+#  define DEAL_II_PETSC_VERSION_GTE(major,minor,subminor) \
+    PETSC_VERSION_GE(major,minor,subminor)
+#endif
```

Comment:
Have these been around for a long time? I believe that they didn't exist when I first wrote the interfaces for PETSc, but that was in 2004/2005. We've long not supported such old versions any more.

Reply:
The only difference is that the PETSc macros also test for `PETSC_VERSION_RELEASE` which is 1 for a release, 0 for any other commit which is not a tag. For example `PETSC_VERSION_LT` will always expand to 0 when developing, while `PETSC_VERSION_GE` will always expand to 1 in the same case.